ArduPlane/navigation.cpp

```diff
@@ -53,7 +54,24 @@ void Plane::loiter_angle_update(void)
     loiter_delta_cd = wrap_180_cd(loiter_delta_cd);
     loiter.sum_cd += loiter_delta_cd * loiter.direction;
 
-    if (labs(current_loc.alt - next_WP_loc.alt) < 500) {
+    /*
+      if doing terrain following then check against terrain target
+     */
+    bool reached_target_alt = false;
+#if AP_TERRAIN_AVAILABLE
+    float altitude_agl;
+    if (reached_target && target_altitude.terrain_following &&
+        terrain.status() == AP_Terrain::TerrainStatusOK &&
+        terrain.height_above_terrain(altitude_agl, true) &&
+        fabsf(altitude_agl - target_altitude.terrain_alt_cm*0.01) < 5) {
```

Comment:
Something doesn't look right here.

In the case that we are flying terrain and we are able to retrieve a terrain-altitude *and* we haven't yet reached that terrain altitude.... we then go and check to see if we have reached our target amsl.  Surely we should accept we haven't reached the terrain altitude yet and keep flying up or down?

i.e. we need another "if" here and one less `&&`.